src/app.component.tsx

```diff
@@ -33,13 +35,38 @@ const assets: Assets = {
 
 export default class App extends React.Component {
 
+  private getActiveRouteName = (navigationState): string | null => {
+    if (!navigationState) {
+      return null;
+    }
+    const route = navigationState.routes[navigationState.index];
+    if (route.routes) {
+      return this.getActiveRouteName(route);
+    }
+    return route.routeName;
+  };
+
+  private onNavigationStateChange = (prevState: NavigationState,
+                                     currentState: NavigationState): void => {
+
+    const prevStateName: string | null = this.getActiveRouteName(prevState);
+    const currentStateName: string | null = this.getActiveRouteName(currentState);
+    if (prevStateName !== currentStateName) {
+      trackScreenTransition(currentStateName)
+        // .then(() => console.log('success'))
+        .catch((error: any) => console.warn('Analytics error: ', error.message));
```

Comment:
wrap this into a function like `onTransitionTrackError`

Reply:
Done.